robot-core/src/main/java/org/obolibrary/robot/template/Template.java

```diff
@@ -0,0 +1,1852 @@
+      String id = null;
+      if (idColumn != -1) {
+        try {
+          id = row.get(idColumn);
+        } catch (IndexOutOfBoundsException e) {
+          // ignore
+        }
+      }
+
+      String label = null;
+      if (labelColumn != -1) {
+        try {
+          label = row.get(labelColumn);
+        } catch (IndexOutOfBoundsException e) {
+          // ignore
+        }
+      }
+
+      if (idColumn != -1 && id == null) {
+        continue;
+      }
+
+      if (id == null || label == null) {
+        continue;
+      }
+
+      String type = null;
+      if (typeColumn != -1) {
+        try {
+          type = row.get(typeColumn);
+        } catch (IndexOutOfBoundsException e) {
+          // ignore
+        }
+      }
+      if (type == null) {
+        type = "class";
+      }
+
+      IRI iri = ioHelper.createIRI(id);
+      if (iri == null) {
+        iri = IRI.create(id);
+      }
+
+      OWLEntity entity;
+      switch (type) {
+        case "":
+        case "owl:Class":
+        case "class":
+          entity = dataFactory.getOWLEntity(EntityType.CLASS, iri);
+          break;
+
+        case "owl:ObjectProperty":
+        case "object property":
+          entity = dataFactory.getOWLEntity(EntityType.OBJECT_PROPERTY, iri);
+          break;
+
+        case "owl:DataProperty":
+        case "data property":
+          entity = dataFactory.getOWLEntity(EntityType.DATA_PROPERTY, iri);
+          break;
+
+        case "owl:AnnotationProperty":
+        case "annotation property":
+          entity = dataFactory.getOWLEntity(EntityType.ANNOTATION_PROPERTY, iri);
+          break;
+
+        case "owl:Individual":
+        case "individual":
+        case "owl:NamedIndividual":
+        case "named individual":
+          entity = dataFactory.getOWLEntity(EntityType.NAMED_INDIVIDUAL, iri);
+          break;
+
+        case "owl:Datatype":
+        case "datatype":
+          entity = dataFactory.getOWLEntity(EntityType.DATATYPE, iri);
+          break;
+
+        default:
+          // Assume type is an individual (checked later)
+          entity = dataFactory.getOWLEntity(EntityType.NAMED_INDIVIDUAL, iri);
+          break;
+      }
+      checker.add(entity, label);
+    }
+  }
+
+  /** Create a Manchester Syntax parser from the OWLDataFactory and QuotedEntityChecker. */
+  private void createParser() {
+    this.parser = new ManchesterOWLSyntaxClassExpressionParser(dataFactory, checker);
+  }
+
+  /**
+   * Process each of the table rows. First, get an entity based on ID or label. If the template
+   * contains an ID column, but it is empty, skip that row. If it does not contain an ID column,
+   * skip if the label is empty. Add axioms based on the entity type (class, object property, data
+   * property, annotation property, datatype, or individual).
+   *
+   * @throws Exception on issue creating axioms from template
+   */
+  private void processRows() throws Exception {
```

Comment:
As a matter of style, I would prefer putting this logic in a singular `processRow(...)` method.